packages/core/src/core/client.ts

```diff
@@ -164,8 +164,17 @@ export class GeminiClient {
     return this.getChat().getHistory();
   }
 
-  setHistory(history: Content[]) {
-    this.getChat().setHistory(history);
+  setHistory(history: Content[], stripThoughts = false) {
```

Comment:
nit: make stripThoughts a named parameter  so its meaning is obvious at callsites.